<h1 style="text-align:center">Comprehensive Exam</h1>

<p>Title Candidates:</p>

<ul>
	<li>Adaptation and its Mechanisms</li>
	<li>Adaptation, Evolvability and Evolution</li>
	<li>Mechanisms of Change and the Evolution of Evolvability</li>
	<li>Ch-ch-ch-changes: Environmental and Genetic Variation Drive the Evolution of Evolvability</li>
	<li>Ch-ch-ch-changes: Environmental and Genetic Change Drive the Evolution of Evolvability</li>
	<li>Environmental and Genetic Change Drive the Evolution of Evolvability</li>
</ul>

<h2 style="text-align:center">Rosangela Canino-Koning</h2>

<h3 style="text-align:center">Dec 26, 2014</h3>

<p>&nbsp;</p>

<h2>1 Introduction (Background)</h2>

<ul>
	<li>Change is the driver for greater and greater adaptation and evolvability. Environmental change, genetic change, and recombination.</li>
	<li>While in static environments, organisms evolve to higher and higher fitnesses as they adapt to their environments[citation to Dave and Mike&#39;s papers]. However, fitness increase does not correlate with evolvability.[reference]</li>
	<li>What is Evolvability?
	<ul>
		<li>Evolvability is the ability of an organism to adapt to a new environment. It can result from
		<ul>
			<li>standing genetic variation in a population (variation)[reference] or (diploidy)[reference], or</li>
			<li>variation of expression within a genome (plasticity) [reference], or</li>
			<li>adaptability of genetic architecture (expandable gene sections) [reference] (modularity) [reference].</li>
			<li>adaptability of behavior (learning) [reference]</li>
		</ul>
		</li>
		<li>Evolvability can be measured by the speed and efficiency by which a population adapts to a new environment. (what are other people doing when they measure evolvability)?
		<ul>
			<li>Measure fitness increase over time (delta fitness/time) (the fitness slope)</li>
			<li>Measure number of mutational steps to adapt to a task</li>
		</ul>
		</li>
		<li>The Definition of evolvability
		<ul>
			<li>Parameters
			<ul>
				<li>Terms should include the organisms/population individual</li>
				<li>What the environmental context term</li>
				<li>The length of the time frame
				<ul>
					<li>short-term evolvability
					<ul>
						<li>Example is plasticity having different effects
						<ul>
							<li>STUDY THIS</li>
						</ul>
						</li>
					</ul>
					</li>
					<li>long-term evolvability</li>
				</ul>
				</li>
				<li>What&#39;s the result for a dead organism? Should be undefined</li>
			</ul>
			</li>
			<li>Final Fitness/Starting Fitness (over some period time)</li>
		</ul>
		</li>
	</ul>
	</li>
	<li>Robustness is a bad thing for evolvability? (Lauren Meyers - Successful professor at UT Austin - part of beacon)
	<ul>
		<li>Robustness lets you move around the genome landscape for free!</li>
		<li>BUT</li>
		<li>This depends on the fitness landscapes!
		<ul>
			<li>Neutral Roads (ridges of neutral mutations that you can use to reach different areas of vs Parking Lots (neutral plateaus that you tend to stay away from the edges of)</li>
		</ul>
		</li>
		<li>Plasticity</li>
	</ul>
	</li>
	<li>David Sloan Wilson
	<ul>
		<li>Group selection guy - seven page thesis</li>
	</ul>
	</li>
	<li>Genomic diffusion rate
	<ul>
		<li>Probability of an offspring being different from parent, but not worse</li>
		<li>Sum of probability of positive or neutral mutations.
		<ul>
			<li>LOOK AT IT ZOMG</li>
			<li>in Charles&#39; thesis
			<ul>
				<li>also one of first two papers</li>
			</ul>
			</li>
		</ul>
		</li>
	</ul>
	</li>
	<li>Evolvability, Sexual Selection, and HGT</li>
	<li><strong>How does change (environmental change, genetic change, mediated genetic change through sex (sexual selection)) promote evolvabiity?</strong></li>
	<li>Definitions of Terms
	<ul>
		<li>What does modularity mean conceptually? How are we measuring it.
		<ul>
			<li>Context, how have other people done it? Lit Review.</li>
		</ul>
		</li>
		<li>What does evolvability mean conceptually? How are we measuring it.
		<ul>
			<li>Context, how have other people done it? Lit Review.</li>
		</ul>
		</li>
	</ul>
	</li>
	<li>We&#39;re using a digital evolution system: Avida</li>
</ul>

<h2>2 Environmental Change, Adaptation and Evolvability</h2>

<h3>[Evolution of Modularity and Evolvability in Changing Environments]</h3>

<h3>2.1 Background</h3>

<p>Genetic modularity contributes to both the robustness and evolvability of natural genomes by allowing disparate functional components to evolve independently of each other, and allowing for the composition of those components into robust gene regulatory networks [1]. However, the evolutionary origins of biological modularity are not well understood. While modularity of functional components is a commonly recurring motif in biological genomes, it appears to be a costly adaptation, which tends to evolve away under static environmental conditions [2]. Even so, research has show that sexual reproduction appears to increase the level of modularity in digital organisms (linear, self- replicating computer programs), as do changing environments that reward modular tasks in evolving neural networks [2].</p>

<p>Under static environmental conditions, where selective pressures and environmental conditions remain fixed, digital organisms have a strong selective pressure to evolve tightly condensed and efficient genomes that exhibit a high degree of pleiotropy and epistasis. Related functional regions are condensed and overlapped, to achieve the execution of required functions with a minimal set of instructions. However, under changing environments, the balance of selective pressures shifts toward favoring evolvability and adaptability rather than pure efficiency. Our research explores whether these shifting pressures result in dif- ferent genetic architectures than what evolves under static conditions, and whether these architectures are more modular, and indeed more evolvable and robust than their static counterparts.</p>

<h3>2.2 Methods</h3>

<p>We used the Avida digital evolution platform to examine the effects of changing environments on the genomes of evolved digital organisms. Digital organisms are circular, self-replicating, and evolving computer programs. Populations of digital organisms are seeded from a single self-replicating ancestor into a 60x60 toroidal world, where the indi- viduals compete with their peers for space to replicate. The organisms that can replicate the fastest come to dominate the environment. In order to replicate faster, the organisms may perform certain rewarded logical operations to increase the speed with which they execute their genomes. They may also shorten and streamline their genomes to reduce the number of instruction executions needed to replicate. The individual genomes are held at a fixed length of 121 instructions, but are mutated after each successful replication event at a rate of 0.00075 point mutation events per site.</p>

<p>We subjected a total of 150 independent populations of digital organisms to two different types of two-phase cyclical changing environments: a benign changing environment, and a hostile changing environment, plus a static (non-changing) environment as a control (fig. 1).</p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/environment.png" style="border-style:solid; border-width:0px; height:200px; width:478px" /></p>

<p style="margin-left:40px">Figure 1: Fluctuating Task Reward in Control, Benign and Hostile Changing Environ- ments.</p>

<p>The benign environmental treatment is characterized by fluctuating reward for per- forming the logical task EQU (the fluctuating task). This reward changes periodically in two phases, from zero reward, to a 25 merit bonus (which represents a multiplier on base organismal fitness of 25), and back to zero. The phases are of equal length, and repeat cyclically throughout the experiment. The organisms are also rewarded continually for performing a different logical task XOR (the backbone task), as a way of encouraging the organisms to preserve sections of their genome through the fluctuating periods, and thus have a basis for comparing the separation or intertwining of functional genetic components. The backbone task is continually rewarded by a 23 merit bonus. Each cycle lasts 1000 updates (roughly 30 generations), and the experiment lasts for 200 cycles per experimental run.</p>

<p>The hostile treatment is identical to the benign treatment except that the reward cycles between a -25 merit bonus (which acts as a strong punishment for performing the task during the off-cycle), and a positive 25 merit bonus.</p>

<p>The static environmental treatment rewards both tasks continuously at 25 (fluctuating task) and 23 (backbone task) merit bonuses.</p>

<h3>2.3 Results and Discussion</h3>

<p>Our experiments show that digital organisms that evolve in cyclic changing environments differ significantly from those evolved in static environments in a number of key ways.</p>

<h4>2.3.1 Evolutionary history and genetic architecture</h4>

<p>Organisms evolved in the hostile changing environment have a much longer phylogenetic history than those evolved in static or benign environments. At each environmental shift, adaptive mutations rapidly sweep and fix in the populations. (figs. 2, 3)</p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/flame_graph__control.png" style="height:153px; width:300px" /></p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/flame_graph__benign.png" style="height:153px; width:300px" /></p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/flame_graph__hostile.png" style="height:153px; width:300px" /></p>

<p style="margin-left:40px">Figure 2: Phylogenetic depth of population evolved in the static (control) environment vs that of hostile-evolved. Sweeps are marked by white horizontal lines. Control-evolved has a maximum depth of 400 ancestors, while hostile-evolved has upward of 1100.</p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/by_site_entropy__control.png" style="border-style:solid; border-width:0px; height:157px; width:300px" /></p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/by_site_entropy__benign.png" style="border-style:solid; border-width:0px; height:157px; width:300px" /></p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/by_site_entropy__hostile.png" style="border-style:solid; border-width:0px; height:157px; width:300px" /></p>

<p style="margin-left:40px">Figure 3: Frequent sweeps resulting from harsh environmental shifts result in very low population and per-site entropy.&nbsp;</p>

<p style="margin-left:40px">&nbsp;</p>

<p>These rapid shifts result in qualitatively different architectural styles. Due to the continuous acceptance of adaptive mutations, the genomes evolved under both experimental treatments are much more scattered, with the bulk of the sites responsible for performing the fluctuating task separated from the backbone task, except for a core region of overlap, which represent portions of the tasks that are shared between XOR and EQU. (fig. 4)</p>

<p>&nbsp;</p>

<p style="margin-left:40px; text-align:center"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/lineagemap__control.png" style="border-style:solid; border-width:0px; height:225px; width:300px" /></p>

<p style="margin-left:40px; text-align:center"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/lineagemap__benign.png" style="border-style:solid; border-width:0px; height:225px; width:300px" /><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/lineagemap__hostile.png" style="border-style:solid; border-width:0px; height:225px; width:300px" /></p>

<p style="margin-left:40px">Figure 4: Sites responsible for performing the tasks are marked, as well as vestigial sites, that is, sites that in a previous generation were responsible for performing a task, but were disabled via mutation elsewhere.</p>

<p style="margin-left:40px">&nbsp;</p>

<p>Interestingly, the treatment-evolved organisms also show development of a large reservoir of vestigial neutral sites, that is, sites that were previously active in performing a task, but were disabled by a mutation elsewhere. These sites appear to be important for allowing the organisms to quickly adapt to the fluctuations in the environment. (fig. 5)</p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/functional_site_count__intertwined.png" style="border-style:solid; border-width:0px; height:126px; width:300px" /></p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/vestigial_site_count__intertwined.png" style="border-style:solid; border-width:0px; height:126px; width:300px" /></p>

<p style="margin-left:40px">Figure 5: Number of functional vs vestigial sites by treatment. The hostile environment has a much larger number of vestigial sites compared to the benign or control, while having a comparable number of functional sites.</p>

<h4>2.3.2 Nearby mutational landscape</h4>

<p>These reservoirs of formerly functional vestigial sites have significantly shifted the treatment-evolved organisms&rsquo; location in genotype space such that losing and re-gaining the fluctuating task is much more often within one or two mutational steps. In this way, the organisms have a significant advantage over the controls in short-term evolvability of the fluctuating task. (fig. 6)</p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/singlestep_lostfluct.png" style="border-style:solid; border-width:0px; height:225px; width:300px" /></p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/evomod/secondstep_regainedfluct.png" style="border-style:solid; border-width:0px; height:225px; width:300px" /></p>

<p style="margin-left:40px">Figure 6: The proportion of mutants that lost or re-gained the task shows the facility with which, in the short-term, the ancestor can adapt to the fluctuating environment.</p>

<h3>2.4 Conclusions</h3>

<p>We conclude that changing environments can indeed produce a kind of functional modularity in digital organisms by separating out unrelated portions of the genome in order to be able to more quickly gain and lose a task. This occurs as a result of the cycles of environmental shift increasing the number of mutations the genomes must accept creating a reservoir of usable neutral mutations. However, it is not clear that this kind of genetic organization is sufficient to produce better long-term evolvability or evolutionary potential.</p>

<p>&nbsp;</p>

<h2>3 Genetic Change via Horizontal Gene Transfer and effect on Evolvability (Proposal)</h2>

<h3>[Horizontal Gene Transfer, Modularity and Evolvability]</h3>

<h3>3.1 Background</h3>

<ul>
	<li>Natural competence</li>
	<li>Mechanisms for transporting DNA fragments</li>
	<li>Rosie Redfield&#39;s work
	<ul>
		<li>Combination of DNA as food, and mechanisms for DNA repair.</li>
	</ul>
	</li>
</ul>

<h3>3.2 Methods</h3>

<ul>
	<li>HGT In Avida
	<ul>
		<li>Genome fragments accumulate as consumable resource in the world
		<ul>
			<li>Fragments decay over time</li>
		</ul>
		</li>
		<li>Fragments may be absorbed via new instructions
		<ul>
			<li>Uptake-HGT-Bonus
			<ul>
				<li>May convey energy OR</li>
				<li>May be integrated into the genome</li>
			</ul>
			</li>
			<li>Uptake-HGT-noBonus
			<ul>
				<li>May be integrated into genome</li>
			</ul>
			</li>
			<li>Uptake-noHGT-Bonus
			<ul>
				<li>Conveys energy only</li>
			</ul>
			</li>
		</ul>
		</li>
	</ul>
	</li>
	<li>Hypothesis 1a: Organisms will uptake if there is energy boost, even with mutational risk.</li>
	<li>Hypothesis 1b: Without potential energy boost, little to no uptake.</li>
	<li>Hypothesis 1c: Evolved organisms will avoid uptake if other methods available.</li>
	<li>Hypothesis 2: Genome fragments with shorter tasks are integrated more often than longer fragments</li>
	<li>Hypothesis 3: Organisms evolved with HGT are more modular than those evolved without HGT</li>
</ul>

<h2>4 Sex and Evolvability</h2>

<h3>4.1 Background</h3>

<ul>
	<li>What&#39;s the difference between HGT and Sex?
	<ul>
		<li>Mechanisms for controlling how genetic material is integrated</li>
	</ul>
	</li>
	<li>How does sex affect evolvability?
	<ul>
		<li>Sex emulates a higher mutation rate?</li>
		<li>Sex without sexual selection increases offspring variation by circulating mutations more widely
		<ul>
			<li>a mutation can hit two parents children instead of just a single parent.</li>
			<li>Sex has downsides! Indiscriminate mating can lead to a much higher rate of genetic&nbsp;variation in offspring than asexual. This is fine in flat fitness landscapes, but not so great in rugged landscapes.
			<ul>
				<li>reference survival of the flattest</li>
			</ul>
			</li>
		</ul>
		</li>
		<li>Standing variation?</li>
	</ul>
	</li>
</ul>

<h3>4.2 Methods</h3>

<ul>
	<li>Measure time to mutational spread with sex vs asexual.</li>
	<li>Neutral mutation is seeded in a population. Measure the number of generations it takes to &nbsp;to spread to&nbsp;everyone in SEXUAL vs ASEXUAL
	<ul>
		<li><em><strong>Surely someone has already done this experiment?</strong></em></li>
	</ul>
	</li>
</ul>

<h2>5 Genetic Change Through Sex, Mediated by Sexual Selection, and the effect on Evolvability</h2>

<h3>5.1 Background</h3>

<h4>5.1.1 Persistence of Sexual Selection</h4>

<ul>
	<li>Why is sexual selection persistent? Is it a runaway process, or does it contribute in some way?</li>
</ul>

<p>There are competing theories about the effects of sexual selection on population diversity and evolutionary potential. The Lek paradox suggests that sexual selection should act against population diversity by restricting the effective population to those phenotypes that display the selected trait. This restriction should result in reduced diversity in the trait, thus minimizing both genetic diversity, and reducing the effectiveness of female choice. However, not only is genetic diversity maintained in biological populations, but female choosiness persists. [RoweHoule1996]</p>

<p>There are several potential resolutions to the Lek paradox. First is genic capture, where condition-dependent sexually selected traits capture the genetic variation surrounding improved condition, and thus that variation is included when high-condition individuals reproduce.[RoweHoule1996]</p>

<p>A second possible explanation for the persistence of female choice is selection equilibrium <strong><em>[find the actual name]</em></strong> where only a low percentage of female choice is required to fuel runaway selection toward displaying a sexually trait.<strong><em>[find a reference]</em></strong></p>

<p>A third possible explanation is that of mutational load, such that mutations are continuously replenishing the diversity of a population.<strong><em>[find the reference. there is one, oddly enough, just gotta find it again.]&nbsp;</em></strong>This diversity creates a fitness gradient that is thus beneficial for mates to select against. <strong>(??)</strong></p>

<h4>5.1.2&nbsp;Ecological Adaptation and Speciation via Sexual Selection</h4>

<ul>
	<li>Sexual selection + condition dependence is a way of locking in traits that are adaptive, so you don&rsquo;t mate with organisms that don&rsquo;t&nbsp;share those adaptive traits. In a way, sexual selection is a way of limiting the damage caused by sex, while still enjoying its benefits.</li>
</ul>

<h4>5.1.3&nbsp;Evolutionary Potential</h4>

<ul>
	<li>If sex changes the architectures of genomes to be more robust (cite Charles&#39;s student argh), and sexual selection locks in adaptations, then shouldn&#39;t sexual selection increase the rate of adaptation as well as improve genomic mobility and genomic diffusion, and decrease pigeonholing? Or do those forces work in opposition to each other? Sex improves structures (because it HAS TO, out of desperation). Sexual selection decreases the mutational load of sex. So, there&#39;s less pressure to be modular and robust because there&#39;s less of an assault going on. So, sexual selection may very well be resulting in pigeonholing because rapid adaptation (due to double-strong selection)&nbsp;+ decreased effective population size/decreased variation/decreased mutational load =&gt; pigeonholing.
	<ul>
		<li>READ ABOUT CANALIZATION</li>
	</ul>
	</li>
	<li>Does sexual selection decrease the effective population size compared to sexual, non-selecting populations? Sexual selection is a way of mutation control? So,&nbsp;</li>
</ul>

<h3>5.2 Proposal - Evolvability and Sexual Selection</h3>

<h4>5.2.2 Methods</h4>

<p>To differentiate between the possibilities, I will set up a series of Avida experiments, with populations of sexual organisms where the females are allowed to choose their mates in the birth chamber based on a condition-dependent trait that the males are allowed to express. I will measure the genetic and phenotypic diversity of the populations over time, in relation to trait expression and female choosiness.</p>

<p>Control: Mates will be assigned randomly, regardless of traits and choosiness.</p>

<p>Genic Capture: Mates will be assigned per female choice. If genetic diversity reduces compared to the control, genic capture can be disqualified.</p>

<p>Runaway Selection Equilibrium: Mates will be assigned per female choice, with a sweep of accuracy percentages. If incidence of the male trait decreases commensurate with reduction in effectiveness of female choice, runaway selection equilibrium can be disqualified.<span style="background-color:#FFFF00"> Ugh, when did I write this? I don&#39;t know what it means.</span></p>

<p>Mutational load: As compared to the control, population diversity should not decrease when sexual selection is in effect.</p>

<h3>5.3 Adaptation and Speciation Via Sexual Selection</h3>

<h4>&nbsp;</h4>

<h4>5.2.2 Methods</h4>

<p><span style="background-color:#FFFF00">Ok, the loghorhea&nbsp;method.</span></p>

<p>What did I do:</p>

<ul>
	<li>General Experimental Structure:
	<ul>
		<li>sexual selection by the females.</li>
		<li>mate choice made based on an integer (trait) that the males display</li>
		<li>tasks structures vary
		<ul>
			<li>a subset of a set of tasks are rewarded.</li>
			<li>sometimes the tasks have prerequisites</li>
		</ul>
		</li>
		<li>allow to evolve in environment with a set of rewarded tasks</li>
		<li>in treatments, split populations into two. allow to evolve in separate environments with different task sets.</li>
		<li>put back together, measure hybridization rate. measure offspring viability. measure offspring&nbsp;fitness.</li>
	</ul>
	</li>
	<li>Limitations
	<ul>
		<li><strong>fitness differential is negligible because of task reward structure</strong>. need to redo experiments because the gradient of selection may be too weak, resulting in randomness in who survives to reproduce.
		<ul>
			<li><strong>How do I compare the hybrid fitness of offspring whose parents have vastly different fitnesses?</strong>&nbsp;<strong>Average the parent log-fitness (thus adjusting for power-based task reward) and if child log-fitness is below the average</strong>, well, there you go? Does that make sense?</li>
		</ul>
		</li>
		<li>better way to test is to&nbsp;<strong>allow them to evolve in the normal way, with normal exponential rewards</strong>. then&nbsp;<strong>calculate the hybridization rate via static analysis</strong>.&nbsp;
		<ul>
			<li>I did some work on this by trying to model the statistical overlap of the traits in R/python (pandas) via iPythonNotebook. I think it was left hanging. I was trying to fit the hybridization rate against the predicted rate based on the trait overlap. If this fitting shows a high correlation, then I can apply the fit to predict hybridization when it&#39;s otherwise impractical to do so because of (above bullet) vastly different fitnesses result in some weird effects when it comes time to mate.</li>
			<li>Or is it just better to&nbsp;<strong>mate everybody with everybody else, and measure the offspring fitness</strong>&nbsp;(as per above averaging method)?</li>
		</ul>
		</li>
		<li><strong>Is it hard to quick run the above experiments?</strong>&nbsp;Um? Not hard? Maybe? I know I got stuck with the work because I went in to R. So, I&#39;m going to avoid that for now. Just do the experiments and see what comes out.&nbsp;<strong>Experiments should be easy, take no more than a couple days.&nbsp;</strong></li>
		<li><strong>Um, I need to figure out which task regime to tweak (see which one was most representative).</strong>&nbsp;One thing fish guy was excited about was that the result (better hybridizing) didn&#39;t seem to vary based on task reward structure. That drift was just as effective. I&#39;m not sure that&#39;s still the case if I bring the task rewards back into the real world, but if it&#39;s still true, then it would be an interesting result, wouldn&#39;t it? So, as a separate experiment, run the assays using the regular reward structure in all the old experimental environments A-E.
		<ul>
			<li><strong>Ok, just drift for now.</strong></li>
		</ul>
		</li>
		<li><strong>Simulate hybridization using static analysis model</strong>
		<ul>
			<li>I need to find the script to do this. It&#39;s in my notebook somewhere.</li>
		</ul>
		</li>
		<li><strong>Do forced hybrid matings and measure average log-fitness of offspring</strong>
		<ul>
			<li>Compare against in-group forced matings to see if hybridizing is bad. (post zygotic effect)</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

<p>&nbsp;</p>

<h4>5.2.3 Results and Discussion</h4>

<h3>&nbsp;</h3>

<h3>5.3 Conclusion</h3>

<h2>6 - Conclusion</h2>

<ul>
	<li>Reference to larger context of this question. Other mechanisms of change - coevolution; luis&#39;s work</li>
</ul>

<p>&nbsp;</p>

<h2>References</h2>

<p>&nbsp;</p>

<h4>Changing Environments</h4>

<ul>
	<li>
	<p>[1] &nbsp;Hartwell, Leland H., et al. &rdquo;From molecular to modular cell biology.&rdquo; Nature 402 (1999): C47-C52.</p>
	</li>
	<li>
	<p>[2] &nbsp;Kashtan, Nadav, and Uri Alon. &rdquo;Spontaneous evolution of modularity and network motifs.&rdquo; Proceedings of the National Academy of Sciences of the United States of Amer- ica 102.39 (2005): 13773-13778.</p>
	</li>
</ul>

<h4>HGT Proposal References</h4>

<ul>
	<li>[3] &nbsp;Redfield, Rosemary J. &rdquo;Do bacteria have sex?.&rdquo; Nature Reviews Genetics 2.8 (2001): 634-639.</li>
	<li>
	<p>[4] &nbsp;Lenski, Richard E., et al. &rdquo;The evolutionary origin of complex features.&rdquo; Nature 423.6936 (2003): 139-144.</p>
	</li>
	<li>
	<p>[5] &nbsp;Yin, Yanbin, et al. &rdquo;Genomic arrangement of bacterial operons is constrained by bi- ological pathways encoded in the genome.&rdquo; Proceedings of the National Academy of Sciences 107.14 (2010): 6310-6315.</p>
	</li>
	<li>
	<p>[6] &nbsp;Deem, Michael W. &rdquo;Statistical mechanics of modularity and horizontal gene transfer.&rdquo; Annu. Rev. Condens. Matter Phys. 4.1 (2013): 287-311.</p>
	</li>
	<li>
	<p>[7] &nbsp;Muoz, Enrique, Jeong-Man Park, and Michael W. Deem. &rdquo;Quasispecies theory for horizontal gene transfer and recombination.&rdquo; Physical Review E 78.6 (2008): 061921.&nbsp;</p>
	</li>
</ul>

<h2>&nbsp;</h2>

<h2>&nbsp;</h2>

<h2>&nbsp;</h2>

<h2>Not Part of Text - Ignore</h2>

<h3><em>Rose Original HGT Proposal</em></h3>

<p>We propose to implement HGT within Avida (see Figure 7). As organisms die, their genomes will be accumulate in reservoirs within world-cells. These reservoirs will age- out, with older genomes disappearing from the reservoir as new ones are accumulated. Fragments for uptake will be randomly selected from the reservoir. Two new mutagenic instructions will be created to trigger uptake of fragments. One of the instructions will confer an energy bonus upon uptake, while the other will not. Both instructions will incur a random chance of incorporating the fragment as an insertion into the genome. If the insertion occurs, no energy bonus will be given. The energy bonus level, uptake mutagenic chance, and fragment size will be configurable per experiment. There are two major hypotheses about HGT usage. First, that HGT usage is a random side-effect of grazing for energy sources[ref]. The second is that HGT is an evolved mechanism for increasing evolvability [ref]. We will test these two hypotheses by giving organisms an opportunity use HGT in various contexts that may or may not reward evolvability.</p>

<p>Hypothesis 1: Organisms will uptake gene fragments from the environment at a higher rate if they are useful as an energy source, even if such an uptake increases the chance of mutation. Without an energy bonus, gene fragment uptake will remain at low frequency.</p>

<p>In static environments, evolvability pressures are expected to be low, therefore the pri- mary benefit of gene fragment uptake is expected to be from the energy received. Without this energy bonus, the disruptive effect of integrating random gene fragments into your genome would result in reduced fitness, and thus lower frequency of uptake in the popula- tion.</p>

<p>Hypothesis 2: Organisms will uptake gene fragments from the environment at a higher rate in environments where evolvability is beneficial, such as in changing environments. In static environments, uptake frequency will remain lower.</p>

<p>Hypothesis 2b: In changing environments, organisms will still uptake gene fragments at higher frequencies, even with reduced, or non-existent bonus reward from uptake.</p>

<p>In environments where evolvability pressures are high, such as changing environments, increased mutation rates are selected for, whereas in static environments, high mutation rates are selected against[ref]. HGT-mediated mutations are therefore expected to be selected for in changing environments, vs static environments. Further, in changing envi- ronments, the energy bonus from uptake may not be neccessary to ensure fragment uptake.</p>

<h3><em>Final BEACON HGT Proposal</em></h3>

<p>We propose to implement HGT within Avida (see Figure 1). As organisms die, their genomes will accumulate in reservoirs. As older genomes are displaced by newer ones, fragments for uptake will be available from the reservoir. We will create three new mutagenic instructions that trigger uptake of these fragments. The first of the instructions, Uptake-HGT-Bonus, will allow an organism to ingest a fragment to metabolize for a fitness bonus, with a chance of recombining it into its genome instead, paralleling an existing theory from the biological literature [3]. The second instruction, Uptake-HGT-noBonus, will allow ingesting the fragment, but not metabolizing, while the third instruction, Uptake-noHGT-Bonus, allows metabolizing the fragment, but no chance for HGT. The magnitude of the fitness bonus, the chance of mutagenic uptake, and the fragment size will be configurable per experiment.</p>

<p style="margin-left:40px"><img alt="" src="./14.356.0_-_Comprehensive_Exam/LaTex/figures/hgt/Figure.png" style="border-style:solid; border-width:0px; height:316px; width:300px" /></p>

<p style="margin-left:40px">Figure 1: The Process of HGT (TODO - NOT FINAL TEXT)</p>

<p>&nbsp;</p>

<p>Our first set of hypotheses examine the some of the environmental conditions that might promote HGT</p>

<p>Hypothesis 1a: Organisms will uptake gene fragments from the environment if they provide a fitness benefit, even if such an uptake has a chance of disrupting their genome.</p>

<p>Hypothesis 1b: Without a fitness benefit, gene fragment uptake will remain at low frequency.</p>

<p>Hypothesis 1c: All else equal, evolved organisms will not risk HGT if an equiv- alent alternative is available.</p>

<p>Many microbes exhibit natural competence, in which DNA is ingested from the environment as a growth medium. DNA is a source of carbon, nitrogen, and phosphorous, as well as many high-energy bonds that can be metabolized. This natural competence comes with a potential cost, in that the DNA may recombine with that of the ingesting organism, thereby disrupting important genes or bring- ing in harmful ones. It is typically assumed that the benefit of foreign DNA as an energetic resource outweighs these costs for the organisms that have evolved natural competence [3].</p>

<p>To test this principle, we will add Uptake-HGT-Bonus to the list of instructions that can result from mutations in Avida, as well as nop-X, a control instruction that does not affect the executing organism or the environment when executed. We will test whether Uptake-HGT-Bonus is used at a higher rate in evolved pop- ulations than nop-X, which would indicate that there is an advantage to using it, rather than it existing at low levels due to mutation alone. Conversely, if an instruction is used significantly less frequently than nop-X, it would indicate an active selective disadvantage associated with that instruction.</p>

<p>We will then replace the Uptake-HGT-Bonus instruction with Uptake-HGT- noBonus, which provides the opportunity for HGT without providing a metabolic benefit to the executing organism. We expect to see a lower rate of use of Uptake-HGT-noBonus than we saw of Uptake-HGT-Bonus, potentially no greater than nop-X. If we find that organisms do use Uptake-HGT-Bonus at greater than nop-X frequency, we will then replace Uptake-HGT-Bonus with Uptake-noHGT-Bonus, which provides CPU cycles but has no chance of causing HGT. We will then com- pare the frequency in evolved genomes of Uptake-noHGT-Bonus to the frequency of Uptake-HGT-Bonus. If Uptake-HGT-Bonus is used more often than Uptake- noHGT-Bonus, that would provide strong evidence that there is an advantage to the HGT component of the instruction independent that of the CPU cycle bonus.</p>

<p>Hypothesis 2: Genomic fragments that contain a complete task in a shorter physical length will be more likely to be incorporated by HGT than fragments that complete a task in a longer physical length. Within evolved Avida populations, the genomic length of a segment used to complete a given task can vary substantially [4]. During recombination, sites that are further apart are more likely to become disassociated from each other than are sites closer together. Because our method of HGT involves recombination of a fragment of a donor genome and the recipient?s genome, we predict that donor genomes that encode a given task in a smaller genomic length will be more likely to be incorporated by HGT. For one, smaller genomic lengths needed for a task mean that a higher proportion of fragments of a given length will contain the complete task. For another, even if a fragment contains an entire task, recombination between a fragment and a host genome does not always occur at the edges of a fragment, and thus a more compact task will be less likely to be disrupted by recombination.</p>

<p>To test this hypothesis, we will perform additional experiments where we seed the fragment reservoirs, rather than letting them fill from the current population. The genome fragments in one set of runs will contain compact tasks, while frag- ments in another set of runs will come from organisms with highly dispersed tasks. We will compare the evolved organisms in these runs to each other and to runs using the previous methodologies where reservoirs were filled with fragments of dead organisms.</p>

<p>Hypothesis 3: Activation of HGT mechanisms will result in evolved genomes with a higher degree of physical modularity than populations without HGT.</p>

<p>Bacterial genomes are strikingly modular. In addition to being compact, with low levels of non-coding DNA, bacterial genomes are typically arranged into oper- ons, where several genes required for a particular function are physically grouped together [5]. This differs from most eukaryotic genomes, where many functions require a series of gene products that are encoded by sites scattered throughout the genome. It has been hypothesized that HGT may cause this modularity, be- cause genomes which happen to have the needed genes for a function grouped physically together will be more likely to spread this grouping to other organisms via HGT than organisms in which the needed genes are more dispersed through the genome [6, 7]. We will test this theory by measuring the physical modularity of genomes from environments with HGT and environments without HGT with the same overall mutation rate.</p>

<p>&nbsp;</p>

<ol>
	<li>&nbsp;</li>
</ol>
